README.md

```diff
@@ -265,6 +265,154 @@ end)
+```lua
+function generateRandomString(length)
+    local characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+    local randomString = ""
+
+    for i = 1, length do
+        local randomIndex = math.random(1, #characters)
+        randomString = randomString .. string.sub(characters, randomIndex, randomIndex)
+    end
+
+    return randomString
+end
+
+RegisterNetEvent('qb-doorlock:client:addNewDoorPS', function()
+	canContinue = false
+	hideNUI()
+	if not Config.SaveDoorDialog then doorData = {} end
+	local dialog = exports['qb-input']:ShowInput({
+		header = "Add a new doorlock to your shell!",
+		submitText = Lang:t("general.submit_text"),
+		inputs = {
+			{
+				text = Lang:t("general.doortype_title"),
+				name = "doortype",
+				type = "select",
+				options = {
+					{ value = "door", text = Lang:t("general.doortype_door") },
+					{ value = "double", text = Lang:t("general.doortype_double") }
+				},
+				default = Config.SaveDoorDialog and doorData.doortype,
+			}
+		}
+	})
+	if not dialog or not next(dialog) then canContinue = true return end
+	doorData = dialog
+
+	local propertyId = exports['ps-housing']:GetPropertyId()
+	local randomchars = generateRandomString("4")
+	local uniqueid = "propertyid_" ..propertyId.. "_door_" ..randomchars
+	doorData.configfile = "PS ( " ..PlayerData.citizenid.. " )"
+	doorData.dooridentifier = uniqueid
+	doorData.doorlabel = uniqueid
+	doorData.cid = PlayerData.citizenid
+	doorData.locked = 'true'
+	doorData.distance = "2"
+
+	local identifier = doorData.configfile..' - '..doorData.dooridentifier
+	if Config.DoorList[identifier] then
+		QBCore.Functions.Notify((Lang:t("error.door_identifier_exists")):format(identifier), 'error')
+		canContinue = true
+		return
+	end
+
+	if doorData.configfile == '' then doorData.configfile = false end
+	if doorData.job == '' then doorData.job = false end
+	if doorData.gang == '' then doorData.gang = false end
+	if doorData.cid == '' then doorData.cid = false end
+	if doorData.item == '' then doorData.item = false end
+	if doorData.doorlabel == '' then doorData.doorlabel = nil end
+	if doorData.pickable ~= 'true' then doorData.pickable = nil end
+	if doorData.cantunlock ~= 'true' then doorData.cantunlock = nil end
+	if doorData.hidelabel ~= 'true' then doorData.hidelabel = nil end
+
+	doorData.locked = 'true'
+	doorData.distance = tonumber(doorData.distance)
+	if doorData.doortype == 'door' or doorData.doortype == 'sliding' or doorData.doortype == 'garage' then
+		local heading, result, entityHit
+		local entity, coords, model= 0, 0, 0
+		QBCore.Functions.Notify("Point any Weapon at the Door. Press [E] to confirm it.", 'primary', 30000)
+		while true do
+			if IsPlayerFreeAiming(PlayerId()) then
+				result, entityHit = raycastWeapon()
```

Comment:
can we just use a normal raycast w/o needing a weapon? i can just see this breaking RP emersion if realtors or home owners have to go around pointing guns at doors

Reply:
https://youtu.be/OrFa-f3D9FY
i guess i got it to work <3